utils/python/CIME/XML/env_batch.py

```diff
@@ -282,6 +282,8 @@ def get_batch_directives(self, case, job, raw=False):
                     default = node.get("default")
                     if not raw:
                         directive = transform_vars(directive, case=case, subgroup=job, default=default, check_members=self)
+                    elif default is not None:
+                        directive = transform_vars(directive, case=case, subgroup=job, default=default)
```

Comment:
Case and subgroup should be None (or left unprovided) here. We want transform to only use the default.